### PR TITLE
add 5.1 release note text

### PR DESCRIPTION
**Pull request by nadinequinr3**
Preview: https://nadine.preview.docs.r3.com/en/platform/corda/5.1/release-notes/5.1.0.html

### CODE REVIEW

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 1]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+
+The performance of the retrieval of unconsumed states in the `findUnconsumedStatesByType` query has been improved. As a result, it it is now more efficient to scan the table with an index on `utxo_visible_transaction_state.consumed`.
+
+#### FindUnconsumedByExactType
+
+A new `FindUnconsumedByExactType` function has been added. The performance of this new function is greater than `FindUnconsumedByType` because it uses the index search that retrieves the given exact-type states (not including subclass states).
+
+#### Verification
+
+Verification functionality has been added to a new `TransactionSignatureVerificationService` interface. This enables it to be injected separately in the verification sandbox.
```
Comment (vlajos) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:64]:
```suggestion
#### Transaction Signature Verification

Signature Verification functionality has been added to a new `TransactionSignatureVerificationService` interface. This enables it to be injected separately in the verification sandbox.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 2]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+
+#### Notary Signed Group Parameters
+
+Signed group parameters are now distributed via backchain resolution to allow notary verification for historic transactions.
```
Comment (vlajos) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:68]:
```suggestion
#### Signed Group Parameters

Signed group parameters are now distributed via backchain resolution to allow notary continuity verification for historic transactions.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 3]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+
+### CorDapp Development
+
+#### Logging for Finality Flows
+
+The ID in logs written for finality flows has been made consistent.
+
+#### Group Parameters
+
+A new interface, `GroupParametersLookup`, has been added to the `net.corda.v5.membership` package to enable flows to access `GroupParameters` for further verification.
```
Comment (vlajos) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:52]:
```suggestion
#### Group Parameters

A new interface, `GroupParametersLookup`, has been added to the `net.corda.v5.membership` package to enable flows to access current `GroupParameters` for further verification.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 4]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+
+#### findUnconsumedStatesByType Performance
+
+The performance of the retrieval of unconsumed states in the `findUnconsumedStatesByType` query has been improved. As a result, it it is now more efficient to scan the table with an index on `utxo_visible_transaction_state.consumed`.
```
Comment (vlajos) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:56]: This function gets deprecated with 5.1. Probably, I would not mention it here.

Comment (blsemo) [reply content/en/platform/corda/5.1/release-notes/5.1.0.md:56]: Maybe we shold rather mentioned that it has been deprecated for removal in 5.2? I am not even sure its performance has been improved?

Comment (nadinequinr3) [reply content/en/platform/corda/5.1/release-notes/5.1.0.md:56]: This was added for this ticket: https://r3-cev.atlassian.net/browse/CORE-15504

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 5]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
 ---
 # Corda 5.1.0 Release Notes
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:12]:
```suggestion
# Corda 5.1 Release Notes
```
You have 5.1 everywhere else. Is there a reason it's 5.1.0 here?

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 6]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+
+The ID in logs written for finality flows has been made consistent.
+
+#### Group Parameters
+
+A new interface, `GroupParametersLookup`, has been added to the `net.corda.v5.membership` package to enable flows to access current `GroupParameters` for further verification.
+
+#### findUnconsumedStatesByType Performance
+
+The performance of the retrieval of unconsumed states in the `findUnconsumedStatesByType` query has been improved. As a result, it it is now more efficient to scan the table with an index on `utxo_visible_transaction_state.consumed`.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:56]:
```suggestion
The performance of the retrieval of unconsumed states in the `findUnconsumedStatesByType` query has been improved. As a result, it is now more efficient to scan the table with an index on `utxo_visible_transaction_state.consumed`.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 7]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+The following methods are deprecated in this release:
+
+* `findUnconsumedStatesByType`
+* `useAndRelease`
+
+The `UtxoLedgerTokenStateObserver` interface has been deprecated. Use `UtxoTokenTransactionStateObserver` instead.
+
+### Deployment
+
+#### Pre-install Checks
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:92]:
```suggestion
#### Pre-Install Checks
```
Same as here: https://nadine.preview.docs.r3.com/en/platform/corda/5.1/deploying-operating/deployment/deploying.html#pre-install-checks

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 8]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+Notary virtual nodes no longer require a ledger key when onboarding to an application network. Corda now rejects any registrations of notary virtual nodes that contain ledger keys.
+
+#### Managing Members
+
+`serial-number` is now mandatory when suspending or re-activating a member.
+
+#### REST API Improvements
+
+* A new version of the REST API, v5_1, is included in this release.
+Old endpoints from v1 work in 5.1 but a warning will be written to the REST Worker log and in the HTTP response.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:158]:
```suggestion
Old endpoints from v1 work in 5.1 but a warning is written to the REST Worker log and in the HTTP response.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 9]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+* The `certificates` and `keys` endpoints have been renamed to `certificate` and `key.`
+* The `GET/hsm/{tenantid}/{category}` endpoint now returns an HTTP 404 status if the category does not exist.
+* The `loginname` parameter of the GET user endpoint has been changed to a path parameter.
+* The `members` endpoint now returns the `MemberInfo` for the member on behalf of who the lookup is performed even if the member has a status of suspended. This ensures that a member is able to check their status at any point in time.
+* The upgrade virtual node endpoint `/api/v5_1/virtualnode/{virtualnodeshortid}/cpi/{targetcpifilechecksum}` now takes an optional query parameter, `forceupgrade`. Set this to `true` if a previous upgrade attempt failed. The default value is false.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:163]:
```suggestion
* The upgrade virtual node endpoint `/api/v5_1/virtualnode/{virtualnodeshortid}/cpi/{targetcpifilechecksum}` now takes an optional query parameter, `forceupgrade`. Set this to `true` if a previous upgrade attempt failed. The default value is `false`.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 10]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+* A new endpoint has been added that returns the result of a flow as JSON: `/flow/{holdingidentityshorthash}/{clientrequestid}/result`.
+
+## Fixed Issues
+
+### Custom Ledger Queries
+
+In some instances, ledger custom queries that used paging skipped rows of data if any of the fields in the query where-clause changed.
+This most commonly occurred if the query used the `consumed` field as part of the where clause and one or more of the states in a previously retrieved page were consumed prior to the final page of the query being returned.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:171]:
```suggestion
This most commonly occurred if the query used the `consumed` field as part of the where-clause and one or more of the states in a previously retrieved page were consumed prior to the final page of the query being returned.
```

File: content/en/platform/corda/5.1/release-notes/5.1.0.md [hunk 11]
```diff
@@ -11,6 +11,164 @@ section_menu: corda51
...
+`serial-number` is now mandatory when suspending or re-activating a member.
+
+#### REST API Improvements
+
+* A new version of the REST API, v5_1, is included in this release.
+Old endpoints from v1 work in 5.1 but a warning will be written to the REST Worker log and in the HTTP response.
+* The `certificates` and `keys` endpoints have been renamed to `certificate` and `key.`
+* The `GET/hsm/{tenantid}/{category}` endpoint now returns an HTTP 404 status if the category does not exist.
+* The `loginname` parameter of the GET user endpoint has been changed to a path parameter.
+* The `members` endpoint now returns the `MemberInfo` for the member on behalf of who the lookup is performed even if the member has a status of suspended. This ensures that a member is able to check their status at any point in time.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.1/release-notes/5.1.0.md:162]:
```suggestion
* The `members` endpoint now returns the `MemberInfo` for the member on behalf of whom the lookup is performed even if the member has a status of suspended. This ensures that a member is able to check their status at any point in time.
```

**Review by AnnaK-R3 — DISMISSED** [review]

File: content/en/platform/corda/5.1/release-notes/5.1.0.md (hunk 4 above)
Comment (blsemo) [reply content/en/platform/corda/5.1/release-notes/5.1.0.md]: Sorry, that was a misunderstanding presented by JIRA... one of those is a sub-task of the other, and they really are the same story.
Yes, `findUnconsumedStatesByExactType` replaces `findUnconsumedStatesByType`, we couldn't improve the performance without changing the behaviour.

Comment (nadinequinr3) [reply content/en/platform/corda/5.1/release-notes/5.1.0.md]: Updated 👍

**Review by AnnaK-R3 — APPROVED** [review]

**Pull request merged**